contrib/checkbox-provider-ce-oem/bin/parse_optee_test.py

```diff
@@ -2,32 +2,62 @@
 
 import json
 import argparse
+import shlex
+import re
+from checkbox_support.snap_utils.snapd import Snapd
+from checkbox_support.snap_utils.system import get_gadget_snap
 
 
-def parse_json_file(filename, filter=False):
-    with open(filename, 'r') as file:
+def parse_json_file(filepath, filter=False, xtest=None):
+    with open(filepath, 'r') as file:
         data = json.load(file)
     for test in data:
         if (filter and test['suite'] == 'pkcs11') or \
            (not filter and test['suite'] not in ['pkcs11']):
-            print_test_info(test)
+            print_test_info(test, xtest)
 
 
-def print_test_info(test):
+def print_test_info(test, xtest):
     print("suite: {}".format(test['suite']))
     print("test_id: {}".format(test['test_id']))
-    print("description: {}\n".format(test['test_description']))
+    print("description: {}".format(test['test_description']))
+    print("tool: {}\n".format(xtest))
+
+
+def look_up_xtest():
+    if Snapd().list("x-test"):
+        xtest = "x-test.xtest"
+    elif look_up_gadget() is not False:
+        xtest = look_up_gadget()
+    else:
+        xtest = None
+    return xtest
+
+
+def look_up_gadget():
+    gadget = get_gadget_snap()
+    xtest = False
+    snap = Snapd().list(gadget)
+    for app in snap["apps"]:
+        if app["name"] == "xtest":
+            xtest = ".".join([app["snap"], app["name"]])
+    return xtest
 
 
 def main():
     parser = argparse.ArgumentParser(description='Parse a JSON file.')
-    parser.add_argument('filename', help='The name of the file to parse.')
+    parser.add_argument('-f', '--filepath',
+                        help='The path of the file to parse.')
     parser.add_argument('-p', '--pkcs11',
-                        help='To filter out PKCS11 for the suite'
+                        help='To filter out PKCS11 for the suite.'
                         'field in JSON.',
                         action='store_true')
     args = parser.parse_args()
-    parse_json_file(args.filename, args.pkcs11)
+    xtest = look_up_xtest()
```

Comment:
I don't get the need to make the python script work for two different things.
I would rather have one script to parse the contents of the json file (`parse_optee_test.py`) and another one to look for the name of xtest (`look_up_xtest.py`)